2011-06-27  Ulrich Drepper  <drepper@gmail.com>
 
	[BZ #12350]
	* nscd/aicache.c (addhstaiX):  Restore only RES_USE_INET6
	bit from old_res_options.

diff --git a/nscd/aicache.c b/nscd/aicache.c
index 3190a13..6c8d83a 100644
--- a/nscd/aicache.c
+++ b/nscd/aicache.c
@@ -534,7 +534,7 @@ next_nip:
    }
 
  out:
-  _res.options = old_res_options;
+  _res.options |= old_res_options & RES_USE_INET6;
 
   if (dataset != NULL && !alloca_used)
     {
